### PR TITLE
Sed 2896 yaml source button does not work for plans with keyword versions

### PR DESCRIPTION
**Pull request by iegorov777**

### CODE REVIEW

**Review by david-stephan — CHANGES_REQUESTED** [review]
@iegorov777 in addition to individual comments, it would be nicer if we would support something like following syntax.
Keep the same for "simple" keyword
keyword: "Measurements"
For call keyword with multiple selection criteria
keyword: 
  name: "Measurements"
  version: "1.0.0"

File: step-plans/step-plans-yaml-parser/src/main/java/step/plans/parser/yaml/rules/KeywordSelectionRule.java [hunk 1]
```diff
@@ -157,4 +163,38 @@ public static DynamicValue<String> getFunctionNameDynamicValue(DynamicValue<Stri
...
+                HashMap<String, JsonNode> functionNameAsMap = jsonObjectMapper.readValue(function.getValue(), functionValueTypeRef);
+                JsonNode simpleFunctionName;
+                if (functionNameAsMap.isEmpty()) {
+                    simpleFunctionName = null;
+                } else if (functionNameAsMap.size() > 1) {
+                    // not only function name is used as selection criteria
+                    simpleFunctionName = null;
+                } else {
+                    simpleFunctionName = functionNameAsMap.get(AbstractOrganizableObject.NAME);
+                }
```
[the remark this answers is not in the document]
Comment (iegorov777) [reply step-plans/step-plans-yaml-parser/src/main/java/step/plans/parser/yaml/rules/KeywordSelectionRule.java:198]: @david-stephan 
Rewritten

**Comment by iegorov777** [comment]
@david-stephan 
In last commit I've changed the format in yaml.
Now the following syntax is supported.

```
- callKeyword:
   keyword: "myKeyword"
```

And also the extended one:

```
- callKeyword:
     keyword:
        name: "myKeyword5"
        version: "1.0.0"
```

File: step-controller/step-controller-server/src/main/java/step/core/plans/PlanServices.java [hunk 2]
```diff
@@ -264,7 +268,12 @@ public Response getYamlPlan(@PathParam("id") String id) {
 		StreamingOutput fileStream = new StreamingOutput() {
 			@Override
 			public void write(java.io.OutputStream output) throws IOException {
-				yamlPlanReader.writeYamlPlan(output, plan);
+				try {
+					yamlPlanReader.writeYamlPlan(output, plan);
+				} catch (Exception ex) {
+					log.error("Serialization error", ex);
+					throw ex;
```
[the remark this answers is not in the document]
Comment (iegorov777) [reply step-controller/step-controller-server/src/main/java/step/core/plans/PlanServices.java]: @david-stephan 
Fixed

**Comment by david-stephan** [comment]
PR on branch 24 was merged and cherry pick to master, closing this one without merge

**Pull request closed without merging**